tests/components/withings/test_sensor.py

```diff
@@ -1,748 +1,264 @@
-        data_manager_instance.async_refresh_token = MagicMock()
-        data_manager_instance.check_authenticated = MagicMock()
-
-        nokia_api_mock.return_value = nokia_api_instance
-        data_manager_mock.return_value = data_manager_instance
-        health_sensor_mock.return_value = MagicMock(spec=WithingsHealthSensor)
-
-        async_add_entities = MagicMock()
-        config_entry = ConfigEntry(
-            1,
-            'domain',
-            'title',
-            {
-                const.PROFILE: 'Person 1',
-                const.CREDENTIALS: {
-                    'access_token': 'my_access_token',
-                    'token_expiry': 'my_token_expiry',
-                    'token_type': 'my_token_type',
-                    'refresh_token': 'my_refresh_token',
-                    'user_id': 'my_user_id',
-                    'client_id': 'my_client_id',
-                    'consumer_secret': 'my_consumer_secret'
-                },
-            },
-            'source',
-            'connection_class'
-        )
-
-        await async_setup_entry(
-            hass,
-            config_entry,
-            async_add_entities
-        )
-
-        async_add_entities.assert_called_with(ANY, True)
-
-
-def test_create_sensor_entities_all(hass: HomeAssistantType):
-    """Test entity creation for all."""
-    data_manager = MagicMock(spec=WithingsDataManager)
-    hass.data[const.DOMAIN] = {
-        const.CONFIG: {}
-    }
-
-    entities = create_sensor_entities(hass, data_manager)
-    assert entities
-    assert len(entities) == len(WITHINGS_MEASUREMENTS_MAP)
-
-
-def test_create_sensor_entities_skip(hass: HomeAssistantType):
-    """Test entity creation skipped."""
-    data_manager = MagicMock(spec=WithingsDataManager)
-    hass.data[const.DOMAIN] = {
-        const.CONFIG: {
-            const.MEASURES: [
-                const.MEAS_BODY_TEMP_C
-            ]
-        }
-    }
-
-    entities = create_sensor_entities(hass, data_manager)
-    assert entities
-    assert len(entities) == 1
-    assert entities[0] == WithingsHealthSensor(
-        data_manager,
-        WITHINGS_MEASUREMENTS_MAP[const.MEAS_BODY_TEMP_C]
-    )
+from .common import nokia_sleep_response
+from .conftest import WithingsFactory, WithingsFactoryConfig
 
 
-@pytest.fixture(name='nokia_api')
-def noka_credentials_fixture():
-    """Provide nokia credentials."""
-    api = nokia.NokiaApi.__new__(nokia.NokiaApi)
-    api.get_credentials = MagicMock(
-        return_value=nokia.NokiaCredentials(
-            user_id='USER_ID'
-        )
+def get_entity_id(measure, profile):
+    """Get an entity id for a measure and profile."""
+    return 'sensor.%s_%s_%s' % (
+        DOMAIN,
+        measure,
+        slugify(profile)
     )
-    api.get_measures = MagicMock()
-    api.get_sleep = MagicMock()
-    api.get_sleep_summary = MagicMock()
 
-    return api
 
+async def assert_state_equals(
```

Comment:
This doesn't need to be a coroutine. There's no `await` in the function. Please make it a regular function.